### PR TITLE
Fix size of maskCanvas to draw image masks correctly

### LINKED ISSUE #17147: Rotated image mask is not rendered correctly
Attach (recommended) or Link to PDF file here: [test.pdf](https://github.com/mozilla/pdf.js/files/13053891/test.pdf)

Configuration:
- Web browser and its version: Firefox 118.0.2
- Operating system and its version: Windows 10 Pro 10.0.19045
- PDF.js version: 3.10.86
- Is a browser extension: No

Steps to reproduce the problem:
1. Open the PDF in Firefox.

What is the expected behavior? (add screenshot)
When I open the PDF in Google Chrome or Adobe Acrobat, it looks like this:
after

What went wrong? (add screenshot)
Instead, it looks like this:
before

### PR DESCRIPTION
**Pull request by JT-External**
The current code creates a Canvas for drawing an ImageMask using only the top-left and bottom-right coordinates of the ImageMask. This code cannot allocate an area on the Canvas to draw the entire ImageMask when the ImageMask is rotated. Therefore, I modified the code to use the four corner coordinates of the ImageMask when creating the Canvas. I also adjusted the offset for drawing on the Canvas according to this modification.

This is a fix for #17147 .

### CODE REVIEW

File: src/display/canvas.js [hunk 1]
```diff
@@ -1313,11 +1313,20 @@ class CanvasGraphics {
...
+      [width, height],
+      [0, height],
+      [width, 0],
+    ].map(cord => Util.applyTransform(cord, maskToCanvas));
+    const xList = cords.map(cord => cord[0]);
+    const yList = cords.map(cord => cord[1]);
+    const minX = Math.min(...xList);
+    const minY = Math.min(...yList);
+    const maxX = Math.max(...xList);
+    const maxY = Math.max(...yList);
```
Comment (Snuffleupagus) [inline src/display/canvas.js:1327]: Unless I'm missing something, it looks like this code is re-implementing the following helper?
https://github.com/mozilla/pdf.js/blob/5f0e5609499374c01cb9ee534f2d44bbd2ae14fb/src/shared/util.js#L725-L738

Can't this be written as follows instead?
```suggestion
    const [minX, minY, maxX, maxY] = Util.getAxialAlignedBoundingBox(
      [0, 0, width, height],
      maskToCanvas
    );
```

File: src/display/canvas.js [hunk 2]
```diff
@@ -1313,11 +1313,20 @@ class CanvasGraphics {
...
+    const drawnWidth = Math.round(maxX - minX);
+    const drawnHeight = Math.round(maxY - minY);
```
Comment (Snuffleupagus) [inline src/display/canvas.js:1329]: The old code ensured that these are non-zero, why was that removed?
```suggestion
    const drawnWidth = Math.round(maxX - minX) || 1;
    const drawnHeight = Math.round(maxY - minY) || 1;
```

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
Thanks for the patch; I've left a couple of inline comments below.

Can you please also add a reference-test?
For an example of how that's done, see e.g. https://github.com/mozilla/pdf.js/commit/459d26edec951fcc049928d3a07f8c933a26d3e5#diff-bbeef90df04bff7617f7ef59e3f37cac92192c5169bc500d857bd6fb107fcabd (and below).

**Comment by Snuffleupagus** [comment]
Please squash the commits into one (we don't use separate fixup commits in this projects); see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits

**Comment by JT-External** [comment]
Thank you for your review and feedback. I appreciate your time and effort. 
I have addressed your inline comments and made the necessary changes. I have also added a reference-test as you requested. 

> Unless I'm missing something, it looks like this code is re-implementing the following helper?

I was not aware of its existence. I have updated my code to use it instead of re-implementing it.

> The old code ensured that these are non-zero, why was that removed?

I removed it by mistake. I have restored it in the latest commit. Thank you for catching that error.

Please let me know if you have any further suggestions or questions. Thank you again for your support.

**Review by Snuffleupagus — APPROVED** [review]
r=me, thank you!

**Comment by marco-c** [comment]
Thanks for the contribution!

**Pull request merged**